src/Symfony/Bundle/WebProfilerBundle/Tests/DependencyInjection/WebProfilerExtensionTest.php

```diff
@@ -31,6 +31,9 @@ public static function assertSaneContainer(Container $container, $message = '')
     {
         $errors = array();
         foreach ($container->getServiceIds() as $id) {
+            if ('service_container' === $id) { // to be removed in 4.0
```

Comment:
having to add such check is because there is no way to opt-in in the new behavior of `getServiceIds`, and then we have to skip the private services to avoid deprecations later. This causes issues (see also #22866 which faces the same issue).

Can we implement a hidden argument allowing to opt-in for the new behavior instead, to get only public ids ?